src/main/java/tictactoe/core/players/HumanPlayer.java

```diff
@@ -1,26 +1,18 @@
 package tictactoe.core.players;
 
 import tictactoe.core.Board;
-import tictactoe.core.UI;
+import tictactoe.core.Mediator;
 
-public class HumanPlayer implements Player {
+public class HumanPlayer extends Player {
 
-    private final PlayerSymbol playerSymbol;
-    private final UI ui;
-
-    public HumanPlayer(PlayerSymbol playerSymbol, UI ui) {
+    public HumanPlayer(PlayerSymbol playerSymbol, Mediator mediator) {
         this.playerSymbol = playerSymbol;
-        this.ui = ui;
-    }
-
-    @Override
-    public int chooseNextMove(Board board) {
-        return ui.requestMove(board, playerSymbol);
+        this.mediator = mediator;
     }
 
     @Override
-    public PlayerSymbol getSymbol() {
-        return playerSymbol;
+    public void requestMove(Board board) {
+        mediator.requestMoveFromUI(board, playerSymbol);
```

Comment:
Correct me if I'm wrong, as the user has the object `Mediator` does it means that it has access to ht e function `receivePlayers(Players players)` do you think it should be possible from the HumanPlayer class?

Reply:
That's a good point, the player definitely has access to way to many methods there. Will have a think how I can reduce it's access.

Reply:
@fabientownsend I've had a go removing access by the players using `lambda expressions`, each player can only call one method from the `Mediator` now